### PR TITLE
add xmlenc as a dependency to gem spec

### PR DESCRIPTION
**Pull request by TimFung93**
addresses https://github.com/saml-idp/saml_idp/issues/127

### CODE REVIEW

**Comment by mjobin-mdsol** [comment]
@Zogoo do you understand why this dependencies needs to be added as a runtime dependency ? did you include this PR in your fork ?

**Comment by jphenow** [comment]
It appears this was [intentional](https://github.com/saml-idp/saml_idp/pull/43), but I could go either way. Is it useful to allow folks to _not_ include it?

**Comment by mjobin-mdsol** [comment]
that makes a lot of sense... 

if it isn't required to everyone, it makes sense to keep it optional.
easy enough to add to our app Gemfile

I think documentation could be clearer on this point though.
it says its optional, but does not say we have to add a gem

**Comment by Zogoo** [comment]
@jphenow , @mjobin-mdsol 
I had this issue and fixed in my fork. This issue come out when we use rbenv or rvm and bundle our project only for production (drop test and development) env. So, I guess most of poeples might using gem in original project and not having any issue, that's why they don't have a issue or maybe they are not using encryption maybe. So, I think this is required gem when you encrypt your response.

**Comment by mjobin-mdsol** [comment]
I would be curious to hear what is the use case for someone to implement SAML without encryption?
but I am guessing 95% of the people want encryption, therefore, it might make sense to merge it.
otherwise, I wonder how we can clarify this on the README, as it seems many people have encountered this puzzling problem.

**Comment by Zogoo** [comment]
@mjobin-mdsol about your first question "SAML response without encryption" use case could be when the SP service requires only SAML Assertion in response get signed by public key not whole response get encrypted. Most cases this will allow to us debug response by browser tool. 

About README, I agree with you we may have a some explanation for what used for may help to understand about this gem, but README might not good place to put technical details. It will be nice if we could create wiki page or something like that.

So, about this pull request I would like to have a xmlenc gem, but we might need to consider about the version.

**Comment by Zogoo** [comment]
@jphenow I have seen peoples are creating pull request add this gem as required gem.
[here](https://github.com/saml-idp/saml_idp/pull/155/files#diff-a8c041b5cf108b781a282a162b8ad679bd6ba6fa185595e82b6d8c7845345b8cR49) and [here](https://github.com/saml-idp/saml_idp/pull/154/files#diff-a8c041b5cf108b781a282a162b8ad679bd6ba6fa185595e82b6d8c7845345b8cR49). So, I would suggest to merge this pull request as quick as possible.

**Comment by Zogoo** [comment]
@TimFung93 do you mind if ask change min version ">= 0.7.1" for your changes?

File: saml_idp.gemspec [hunk 1]
```diff
@@ -47,6 +47,7 @@ section of the README.
   s.add_dependency('uuid', '>= 2.3')
   s.add_dependency('builder', '>= 3.0')
   s.add_dependency('nokogiri', '>= 1.6.2')
+  s.add_dependency('xmlenc', '>= 0.6.4')
```
Comment (Zogoo) [inline saml_idp.gemspec:50]: @TimFung93 do you mind if I ask change min version ">= 0.7.1" for your changes?

**Review by Zogoo — CHANGES_REQUESTED** [review]

File: saml_idp.gemspec (hunk 1 above)
Comment (TimFung93) [reply saml_idp.gemspec]: done

**Review by jphenow — APPROVED** [review]

**Pull request merged**